depthai_examples/ros1_src/yolov4_spatial_publisher.cpp

```diff
@@ -125,14 +140,20 @@ int main(int argc, char** argv){
       pnh.getParam("nn_path", nnPath);
     }
 
-    dai::Pipeline pipeline = createPipeline(syncNN, subpixel, nnPath, confidence, LRchecktresh);
-    dai::Device device(pipeline);
+    dai::Pipeline pipeline = createPipeline(syncNN, subpixel, nnPath, confidence, LRchecktresh, monoResolution);
 
+    dai::rosBridge::ImageConverter rgbConverter(deviceName + "_rgb_camera_optical_frame", false);
+    _dev = std::make_unique<dai::Device>(pipeline);
+    auto colorQueue = _dev->getOutputQueue("preview", 30, false);
+    auto detectionQueue = _dev->getOutputQueue("detections", 30, false);
+    auto depthQueue = _dev->getOutputQueue("depth", 30, false);
+    std::shared_ptr<dai::DataOutputQueue> stereoQueue;
+    auto calibrationHandler = _dev->readCalibration();
+
+    //TODO this part would be removed once we have calibration-api
     std::string color_uri = camera_param_uri + "/" + "color.yaml";
-     std::string stereo_uri = camera_param_uri + "/" + "right.yaml";
 
-    dai::rosBridge::ImageConverter rgbConverter(deviceName + "_rgb_camera_optical_frame", false);
-    dai::rosBridge::BridgePublisher<sensor_msgs::Image, dai::ImgFrame> rgbPublish(device.getOutputQueue("preview", 30, false),
+    dai::rosBridge::BridgePublisher<sensor_msgs::Image, dai::ImgFrame> rgbPublish(colorQueue,
```

Comment:
`auto rgbCameraInfo = rgbtConverter->calibrationToCameraInfo(calibrationHandler, dai::CameraBoardSocket::RGB, width, height)`

You can do the same for RGB camera too instead of using color.yaml

Reply:
Thanks.
I apologize for my lack of understanding.
Should the height and width match the preview(416,416) or the 1080p(1920, 1080)?

Reply:
Fixed in b9b6b81. 
CameraInfo match (416, 416). If I misunderstood, please comment.